src/libsystemd-network/dhcp-network.c

```diff
@@ -186,19 +186,20 @@ int dhcp_network_bind_udp_socket(int ifindex, be32_t address, uint16_t port, int
                         return r;
         }
 
+        r = setsockopt_int(s, SOL_SOCKET, SO_BROADCAST, true);
+        if (r < 0)
+                return r;
```

Comment:
can you explain why you moved this?

Reply:
Sending broadcast is required for dhcp-server socket, regardless of how it's bound.

Reply:
Moving this outside of the condition makes dhcp-client socket to have SO_BROADCAST too(unlike current implementation). But I thought that adding check like `if (port == DHCP_PORT_SERVER)`  or additional function parameter will be more confusing then useful.